strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/io/ArtifactOutputStream.java

```diff
@@ -15,18 +20,118 @@
 public class ArtifactOutputStream
         extends MultipleDigestOutputStream
 {
-
+    private Function<byte[], String> digestStringifier = MessageDigestUtils::convertToHexadecimalString;
     private ArtifactCoordinates coordinates;
+    /**
+     * Used to cache Artifact contents if needed.
+     */
+    private OutputStream cacheOutputStream;
```

Comment:
I thought that it would be useful to be able to write Artifact with additional `OutputStream` in addition to the wrapped `OutputStream`. It used to write checksum into `ByteArrayOutputStream` in `ArtifactManagementServiceImpl`.
`OutputStream` does not mean writing into memory.